src/test/util/logging.h

```diff
@@ -16,11 +16,13 @@ class DebugLogHelper
     const std::string m_message;
     bool m_found{false};
     std::list<std::function<void(const std::string&)>>::iterator m_print_connection;
+    using MatchFn = std::function<bool(const std::string*)>;
```

Comment:
```suggestion
    using MatchCb = std::function<void(const std::string&)>;
```

why does this need a return value? From glancing over it, it seems the same can be achieved with a callback that circumvents `m_found` and `m_cb` is set to nullptr by default.

I presume you want to make this as flexible as possible for potential future use cases?